src/shared/components/AppLayout.js

```diff
@@ -0,0 +1,22 @@
+import React, {Component} from 'react';
+import Header from '../components/header';
+
+class AppLayout extends Component {
+  render() {
+    return (
+      <div>
+        <Header className="header"/>
+        <div className="container">
+          {this.props.children}
+        </div>
+        <script src={'/javascripts/main.js'}/>
```

Comment:
I think there is no big difference in speed and I think this is proper way since it is server-side rendering.